docs/developer_manual/services/advanced/service_updater_upgrade.md

```diff
@@ -0,0 +1,269 @@
+    with SigmaUpdateServer(default_pattern="*.yml") as server:
+        server.serve_forever()
+```
+
+## Is there anything I need to change for the actual service code?
+There would be a need to implement a `load_rules()` since services can use their rules in different ways so standardizing on a method isn't feasible at the moment.
+
+In the case of Sigma:
+
+```python
+    def _load_rules(self) -> None:
+        temp_list = []
+        # Patch source_name into signature and import
+        for rule in self.rules_list:
+            with open(rule, 'r') as yaml_fh:
+                file = yaml_fh.read()
+                source_name = rule.split('/')[-2]
+                patched_rule = f'{file}\nsignature_source: {source_name}'
+                temp_list.append(patched_rule)
+
+        self.log.info(f"Number of rules to be loaded: {len(temp_list)}")
+        for rule in temp_list:
+            try:
+                self.sigma_parser.add_signature(rule)
+            except Exception as e:
+                self.log.warning(f"{e} | {rule}")
+
+        self.log.info(f"Number of rules successfully loaded: {len(self.sigma_parser.rules)}")
+```
+
+Note: After each submission, during a `_cleanup()`, there is a call made to `_download_rules()` which will ask your updater if there's new signatures to retrieve. If there's nothing new, then you will proceed to use the same signature set. Otherwise, the service wil retrieve the new set and attempt to load it into the service. For this reason, there can be the need to implement a `_clear_rules()` to ensure service stability if there was a failure while loading the new set that could have a negative impact on analysis.
+
+## What don't I need to implement and why?
+### `do_source_update()`
+Considering most of our services have the same pattern of:
+ 1. Establishing a client for Assemblyline using the service_update_account
+ 2. Iterating over our source list and using either a URL download or Git clone to retrieve signatures
+ 3. (Optional) If there are rules to be updated, modify the rules before import to add extra metadata or validation
+ 4. Importing the rules into Assemblyline
+
+ We decided to streamline that process for most services while giving service writers the ability to override certain aspects as needed.
+
+ The service base contains a set of helper functions for retrieving files during the download process. So you'll notice most services don't need to implement their own `do_source_update()`, they just need to implement an `import_update()`. [Safelist](https://github.com/CybercentreCanada/assemblyline-service-safelist) is an example of a service that is an exception to this rule where it does implement it's own `do_source_update()`.
+
+### `do_local_update()`
+This particular function doesn't need to be overridden (but you're free to do so) because it's primary function is retrieve the signature set from Assemblyline and make it accessible to service instances when they ask for signatures.
```

Comment:
```suggestion
This particular function doesn't need to be overridden (but you're free to do so) because its primary function is to retrieve the signature set from Assemblyline and make it accessible to service instances when they ask for signatures.
```